homeassistant/components/proxmoxve/__init__.py

```diff
@@ -91,100 +94,132 @@
-                proxmox_client = ProxmoxClient(
-                    host, port, user, realm, password, verify_ssl
-                )
-                proxmox_client.build_client()
-            except AuthenticationError:
-                _LOGGER.warning(
-                    "Invalid credentials for proxmox instance %s:%d", host, port
-                )
-                continue
-            except SSLError:
-                _LOGGER.error(
-                    "Unable to verify proxmox server SSL. "
-                    'Try using "verify_ssl: false" for proxmox instance %s:%d',
-                    host,
-                    port,
-                )
-                continue
-            except ConnectTimeout:
-                _LOGGER.warning("Connection to host %s timed out during setup", host)
-                continue
-            except requests.exceptions.ConnectionError:
-                _LOGGER.warning("Host %s is not reachable", host)
-                continue
+    host = entry_data[CONF_HOST]
+    port = entry_data[CONF_PORT]
+    user = entry_data[CONF_USERNAME]
+    realm = entry_data[CONF_REALM]
+    password = entry_data[CONF_PASSWORD]
+    verify_ssl = entry_data[CONF_VERIFY_SSL]
 
-            hass.data[PROXMOX_CLIENTS][host] = proxmox_client
-
-    await hass.async_add_executor_job(build_client)
+    try:
+        # Construct an API client with the given data for the given host
+        proxmox_client = ProxmoxClient(host, port, user, realm, password, verify_ssl)
+        await hass.async_add_executor_job(proxmox_client.build_client)
+    except AuthenticationError as error:
+        raise ConfigEntryAuthFailed from error
+    except SSLError:
+        _LOGGER.error(
+            "Unable to verify proxmox server SSL. "
+            'Try using "verify_ssl: false" for proxmox instance %s:%d',
+            host,
+            port,
+        )
+        return False
+    except ConnectTimeout:
+        _LOGGER.warning("Connection to host %s timed out during setup", host)
+        return False
 
     coordinators: dict[str, dict[str, dict[int, DataUpdateCoordinator]]] = {}
-    hass.data[DOMAIN][COORDINATORS] = coordinators
 
-    # Create a coordinator for each vm/container
-    for host_config in config[DOMAIN]:
-        host_name = host_config["host"]
-        coordinators[host_name] = {}
+    proxmox = await hass.async_add_executor_job(proxmox_client.get_api_client)
 
-        proxmox_client = hass.data[PROXMOX_CLIENTS][host_name]
+    node = entry_data[CONF_NODE]
+    node_coordinators = coordinators[node] = {}
 
-        # Skip invalid hosts
-        if proxmox_client is None:
-            continue
+    # Proxmox instance info
+    coordinator = create_coordinator_container_vm(
+        hass, proxmox, entry_data[CONF_HOST], None, None, ProxmoxType.Proxmox
+    )
 
-        proxmox = proxmox_client.get_api_client()
+    # Fetch initial data
+    await coordinator.async_refresh()
```

Comment:
I would call `async_config_entry_first_refresh` so that if it fails, it will automatically raise `ConfigEntryNotReady`.